client/my-sites/plans-features-main/index.tsx

```diff
@@ -439,12 +446,15 @@ const PlansFeaturesMain = ( {
 		useFreeTrialPlanSlugs,
 	} );
 
-	// we neeed only the visible ones for features grid (these should extend into plans-ui data store selectors)
+	// we need only the visible ones for features grid (these should extend into plans-ui data store selectors)
 	const gridPlansForFeaturesGrid = useGridPlansForFeaturesGrid( {
 		allFeaturesList: getFeaturesList(),
 		coupon,
 		eligibleForFreeHostingTrial,
-		hiddenPlans,
+		hiddenPlans: {
+			...hiddenPlans,
+			hideFreePlan: hideFreePlan || deemphasizeFreePlan,
```

Comment:
The primary concern I have for the current implementation is that this would add a case for sustaining the visibility flags, which contradicts to https://github.com/Automattic/wp-calypso/issues/88403. 

In this case, I need the full plan mix, but want the features grid to not show the Free plan while the comparison grid does. Currently the most intuitive way seems to be using the visibility flags. 

@chriskmnds @oswian Is this a legit use of the flags? I'm totally up for any other alternatives, so please feel free to let me know 🙇🏼

Reply:
I wonder if post-processing `gridPlansForFeaturesGrid` to remove the free plan might work equally (meaning, it might be possible to avoid adding to `hiddedPlans` that way)? Or is that going to mess with the highlight headers/borders? 🤔

Reply:
@ddc22 Great thoughts :) 
Do you mean to feed different plan mixes from different intents to the features grid and the comparison grid? In that case I'm concerning that the underlying plans data might be doubled, since both `useGirdPlansForFeaturesGrid` and `useGridPlansForComparisonGrid` are using the same `useGridPlans` hook under the hood. 

I also would like to introduce the deemphasizing free plan mechanism as one cohesive concept that will work outside of the intent system rather than 2 separated changes, hence binding them by the same prop. That way, whenever someone want to enable the functionality, they will only need to touch one thing. 

Does that make sense to you? 
Since I need to wrap this project up today so I'm going to ship this as it is for now, but please feel free to leave any follow-up thought that you might have 🙇🏼

Reply:
> Sorry if I am a bit too purist here. The prop clean up has instilled in me an intense desire to keep additional props out of the grid as much as possible from the component API 

No worries. I love and appreciate how you think deeper into the very details of the whole data flow :) 

> Aren't these hooks just preprocessing pure functions called consecutively? Not sure how this would double the data since these two structures already excist.

The flow is a bit different now since https://github.com/Automattic/wp-calypso/pull/87478/. Before that, `useGridPlans` was called and then the returned data was then further processed by `useGridPlansForComparisonGrid` and `useGridPlansForFeaturesGrid`. So it was guaranteed that there would be one copy of the original `gridPlans`, and then a processed copy of `gridPlans` for the comparison grid and a processed copy of `gridPlans` for the features grid. Now that `useGridPlans` is "implicitly" called within `useGridPlansForComparisonGrid` and `useGridPlansForFeaturesGrid`, passing in different intent should result in different copies of the original 

However, after writing all these I actually realized that **these are all my assumption of how it works underneath**, instead of thinking at the right abstraction level. 🤔

> Wouldn't we have to remove this in case the experiment favours control?

Actually I want to keep it 😆 If the experiment favors control, I'll just turn it off on the onboarding flow while keeping it for `onboarding-pm`. However, now I can see that your way will work equally without introducing a new prop. 

Thanks for brainstomring with me. I'll revisit this when the experiment concludes. During which, I'll keep it as is to err on the safe side and only address the cache missing issue @chriskmnds [raised below](https://github.com/Automattic/wp-calypso/pull/89395#discussion_r1567315085).

Reply:
I think having separate intents for Features Grid & Comparison Grid should be a bit of a last resort, and definitely not something that we can "just do" in the code - not without rethinking the entire abstraction mechanism. The logic and inferences about intent are meant to be shared on the pricing page, so there needs to be a single/clear way to say `intent === "foo"`. It raises questions when `foo-comparison` and `foo-other` coexist on the same page / at the same time. We'd likely need to localize these types to not spill into conditions in the code, work strictly through selectors, potentially build subclassing, etc.

Reply:
but yeah, my immediate reaction was also multiple intents - from which I backed out from the moment I realized `intent === "foo" || intent === "foo-comparison"` could become an unregulated reality in the code 😆